src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -535,14 +545,66 @@ public Builder onShiftClick(@Nullable ShiftClickAction<?> shiftClickAction) {
         }
 
         /**
+         * Gets if this builder should try to simplify when appended text.
+         *
+         * @return If this builder will simplify when appended text.
+         */
+        public boolean getSimplify() {
+            return this.simplify;
+        }
+
+        /**
+         * Sets if this builder should simply when appending text.
+         *
+         * @param simplify The new simplified state
+         * @return This text builder
+         */
+        public Builder simplify(boolean simplify) {
+            this.simplify = simplify;
+            return this;
+        }
+
+        /**
+         * Appends the specified {@link Text} to the end of this Text,
+         * making sure to simplify if possible.
+         *
+         * @param child The text to append
+         * @return This text builder
+         */
+        public Builder appendSimplified(Text child) {
+            if (!child.isEmpty()) {
+                int lastIndex = this.children.size() - 1;
```

Comment:
Yes next line is a check for if `children` is empty or not. I guess I just forgot to move `lastIndex` into that block.